payments/providers/base.py

```diff
@@ -0,0 +1,118 @@
+from urllib.parse import urlencode
+
+from django.http import HttpResponse, HttpResponseNotFound, HttpResponseServerError
+from django.shortcuts import redirect
+from django.urls import reverse
+
+from ..models import Order
+
+
+class PaymentProvider:
+    """Common base for payment provider integrations"""
+
+    ui_return_url_param_name = "VENE_UI_RETURN_URL"
+
+    def __init__(self, **kwargs):
+        if "config" in kwargs:
+            self.config = kwargs.get("config")
+        self.request = kwargs.get("request")
+        self.ui_return_url = kwargs.get("ui_return_url")
+
+    def initiate_payment(self, order: Order) -> str:
+        """Create a payment to the provider.
+        Implement this in your subclass. Should return a URL to which the user
+        is redirected to actually pay the order."""
```

Comment:
should this `raise NotImplementedError`?

Reply:
well, Tuomas left it out, maybe there was a reason 😅 🤷‍♂️ let's follow respa

Reply:
I had the exact same train of thought. But I guess we can just add it as a failsafe?

Reply:
let's add it, I think it should be ok